pkg/datasource/coinmarketcap/v1/types.go

```diff
@@ -0,0 +1,66 @@
+package v1
+
+import (
+	"time"
+)
+
+type EndpointType string
+
+const (
+	EndpointTypeHistorical EndpointType = "historical"
+	EndpointTypeLatest     EndpointType = "latest"
+	EndpointTypeNew        EndpointType = "new"
+)
+
+type Response struct {
+	Data   []Data `json:"data"`
```

Comment:
you can use json.RawMessage, so that you can use `-responseDataType` option from requestgen.